Assets/Scripts/PathMovement.cs

```diff
@@ -12,13 +12,17 @@ public class PathMovement : MonoBehaviour {
     private GameObject instantiatedPath;
     public float speed = 1.0f;
 
+    private StaminaBar staminaBar;
+
     private int nextPointIndex = -1;
     private bool pathComplete = false;
     // Use this for initialization
     void Start () {
 		if(thingToMove == null) {
             thingToMove = this.gameObject;
         }
+
+		staminaBar = this.gameObject.AddComponent<StaminaBar>() as StaminaBar;
```

Comment:
don't add components through the script, add the component to the prefab, so the `Player.prefab` always have a component. This duplicates the `StaminaBar` component